### PR TITLE
New Zolertia Re-Mote platform (CC2538 + CC1120)

### PR DESCRIPTION
**Pull request by alignan**
This PR brings the new Zolertia Re-Mote platform (TI CC2538SF53 + TI CC1120) into Contiki, featuring:

* ISM 2.4-GHz IEEE 802.15.4 & Zigbee compliant.
* ISM 868-, 915-, 920-, 950-MHz ISM/SRD Band, IEEE 802.15.4g compliant.
* AES-128/256, SHA2 Hardware Encryption Engine.
* ECC-128/256, RSA Hardware Acceleration Engine for Secure Key Exchange.
* Power consumption down to 3uA using our shutdown mode (disables all hardware but Co-Processor, saves energy of connected sensors and actuators as well).
* Co-Processor to allow peripheral energy management, programming over BSL without requiring to press any button to enter bootloader mode.
* Built-in battery charger (500mA), Energy Harvesting and Solar Panels to be connected to standards LiPo batteries.
* Built-in TMP102 low-power digital temperature sensor (1uA shutdown).
* Power input with wide range 2-26VDC.
* Built-in MicroSD for external storage.
* RGB LED.
* User button mapped to PAx to be able to manually unlock the bootloader.
* 2 x micro USB connectors: USB0 for programming/debugging, USB1 is USB 2.0 ready.
* RF switch to allow programatically to select the 2.4Ghz internal (ceramic chip) or external antenna.
* Small form-factor (as the Z1 mote, half the size of an Arduino) 57x35 mm.

Support for CC1120 has already been tested and asserted with long-range field tests (at this [branch](https://github.com/alignan/contiki/tree/remote-pr-contrib-sensors-no-startup-with-cc1120)), pending to be refactored after #1002 is merged.

It has been tested over some months now at our headquarters and amongst early beta-testers, as well as our partners in the FP7 European Project in which this platform was born.  It is commercially available at our website as well, of course I will be its maintainer.

More information at <http://zolertia.io/>

And a fun example of a project using the platform:
<https://www.hackster.io/alinan/wireless-iot-mqtt-super-mario-bros-coin-box>

<a href="url"></a>

### CODE REVIEW

File: platform/remote/dev/led-strip.c [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    return LED_STRIP_ERROR;
+  }
+
+  /* Inverse logic, return ON if the pin is low */
+  if(GPIO_READ_PIN(LED_STRIP_PORT_BASE, LED_STRIP_PIN_MASK)) {
+    return LED_STRIP_OFF;
+  }
+  return LED_STRIP_ON;
+}
+/*---------------------------------------------------------------------------*/
```
Comment (bthebaudeau) [inline platform/remote/dev/led-strip.c]: All files should finish with a newline.

File: platform/remote/dev/phidget-sensor.c [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    channel = SOC_ADC_ADCCON_CH_AIN0 + ADC_PHIDGET_ADC3_PIN;
+    break;
+  default:
+    return REMOTE_SENSORS_ERROR;
+  }
+
+  res = adc_get(channel, SOC_ADC_ADCCON_REF_INT, decimation_rate);
+
+  /* Shift right by a number of bits depending on the decimation rate */
+  return res >> shift;
```
Comment (bthebaudeau) [inline platform/remote/dev/phidget-sensor.c]: It would be better to remove all the ADC shift stuff from this PR, as discussed in #968.

**Comment by alignan** [comment]
Thanks @bthebaudeau! I will make the changes and push an update

**Comment by alignan** [comment]
Addressed all comments and squashed into single commit, waiting for Travis now.

**Comment by bthebaudeau** [comment]
You can drop the changes from `soc-adc.h`.

Apart from that, this PR looks good after a very quick review.

**Comment by g-oikonomou** [comment]
Not hugely keen on the doxygen module structure. Either we should put everything under "The Re-Mote Development Kit platform" or everything under "TI cc2538-powered platforms". Currently the defgroup in board.h misplaces stuff.

Other than that, I've been involved with this platform since it was only an idea :+1:

**Comment by alignan** [comment]
Removed leftovers from `soc-adc.h` and updated doxygen group

**Comment by g-oikonomou** [comment]
Doc structure now looks good!

File: platform/remote/dev/board.h [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+ * \note   Do not include this file directly. It gets included by contiki-conf
+ *         after all relevant directives have been set.
+ */
+#ifndef BOARD_H_
+#define BOARD_H_
+
+#include "dev/gpio.h"
+#include "dev/nvic.h"
+/*---------------------------------------------------------------------------*/
+/** \name SmartRF LED configuration
```
Comment (bthebaudeau) [inline platform/remote/dev/board.h]: Probably `Re-mote`.

File: platform/remote/contiki-main.c [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+ */
+/**
+ * \addtogroup platform
+ * @{
+ *
+ * \defgroup remote Zolertia Re-Mote platform
+ *
+ * The Re-Mote is the new platform by Zolertia based on the cc2538, featuring a
+ * cc2530 SoC with an ARM Cortex-M3 core plus the CC1120 Sub-1Ghz transceiver
+ * @{
```
Comment (bthebaudeau) [inline platform/remote/contiki-main.c]: You mean `cc2538`, right?

**Comment by alignan** [comment]
Fixed, thanks!

**Comment by bthebaudeau** [comment]
:+1: and Travis is happy.

**Comment by joakimeriksson** [comment]
Really nice - when do you start shipping? I saw only pre-order in the shop yet?

**Comment by alignan** [comment]
We plan to ship by August, we are to release 1-2 more products as well!

**Comment by joakimeriksson** [comment]
Nice - I will order some as soon as you have them ready to ship!

**Pull request merged**